src/coreclr/jit/emitarm64.cpp

```diff
@@ -12311,15 +12311,11 @@ size_t emitter::emitOutputInstr(insGroup* ig, instrDesc* id, BYTE** dp)
 
                 // If there are 2 GC vars in this instrDesc, get the 2nd variable
                 // that should be tracked.
-                adr2     = emitComp->lvaFrameAddress(varNum2, &FPbased2);
+                adr2     = emitComp->lvaFrameAddress(varNum2, &FPbased2, true);
                 ofs2Dist = EA_SIZE_IN_BYTES(size);
 #ifdef DEBUG
                 assert(FPbased == FPbased2);
-                if (FPbased)
-                {
-                    assert(id->idReg3() == REG_FP);
```

Comment:
i forgot to ask, but why this assert was removed?

Reply:
The optimization in `lvaFrameAddress` can rewrite FP-n expression to SP+m in the instruction encoding which would be one of the inputs here. Since here we call `lvaFrameAddress` again with an additional parameter that suppresses this optimization (to always get FP-relative offset for purpose of GC info offset for something that was FP-relative in the first place), we can can get a mismatch. This mismatch is expected.

(The general idea of the optimization is that in the instruction encoding we can substitute the two because they refer to the same location and positive offsets have wider range that can be expressed in the instruction encoding. For the GC info offsets we want to use the original offsets due to the way how GC info is encoded. This closely matches the same logic in ARM32.)